### PR TITLE
Feature/ Deposit and Withdrawal of House module KYC Validation

### PR DESCRIPTION
**Pull request by scorpioborn**
## Description

Blockchain needs to check for KYC in the house deposit and withdrawal, This feature needs the House module to include the KYC validation in the ticket while processing the deposit and withdrawal message.

---

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Added Features

- [x] Ticket data with KYC in Deposit
- [x] Ticket data with KYC in Withdrawal
- [x] Validate the KYC in Deposit and Withdrawal

---

## Modified Features

- [x] Dead markdown link of the bet spec doc for the module readme
- [x] CLI and Message Validation
- [x] Proto messages of the TXs

---

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
---

### CODE REVIEW

File: proto/sge/house/kyc.proto
Comment (3eyed-raven) [inline proto/sge/house/kyc.proto:1]: This is duplicate proto. Is there a way to keep it globally and reuse it in bet, house module both?

File: x/house/client/cli/tx_deposit.go [hunk 1]
```diff
@@ -15,8 +15,8 @@ import (
 
 func CmdDeposit() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "deposit [market_uid] [amount]",
-		Args:  cobra.ExactArgs(2),
+		Use:   "deposit [market_uid] [amount] [ticket]",
+		Args:  cobra.ExactArgs(3),
 		Short: "Deposit tokens in a market order book to be the house",
 		Long: strings.TrimSpace(
 			fmt.Sprintf(`Deposit tokens in a market order book to be the house.
```
Comment (3eyed-raven) [inline x/house/client/cli/tx_deposit.go:22]: example needs to be updated

File: x/house/client/cli/tx_withdraw.go [hunk 2]
```diff
@@ -17,8 +17,8 @@ import (
 
 func CmdWithdraw() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "withdraw [market_uid] [participation_index] [mode] [amount]",
-		Args:  cobra.RangeArgs(3, 4),
+		Use:   "withdraw [market_uid] [participation_index] [ticket] [mode] [amount]",
+		Args:  cobra.RangeArgs(4, 5),
 		Short: "Withdraw tokens from a deposit",
 		Long: strings.TrimSpace(
 			fmt.Sprintf(`Withdraw coins of unused amount corresponding to a deposit.
```
Comment (3eyed-raven) [inline x/house/client/cli/tx_withdraw.go:24]: example needs to be updated

File: x/house/types/ticket.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package types
+
+import (
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// Validate validates market add ticket payload.
```
Comment (3eyed-raven) [inline x/house/types/ticket.go:7]: comments mention market where as it should be house

File: proto/sge/house/ticket.proto [hunk 4]
```diff
@@ -0,0 +1,20 @@
+
+syntax = "proto3";
+package sgenetwork.sge.house;
+
+import "gogoproto/gogo.proto";
+import "sge/house/kyc.proto";
+
+option go_package = "github.com/sge-network/sge/x/house/types";
+
+// DepositTicketPayload indicates data of bet deposit ticket.
```
[the remark this answers is not in the document]
Comment (scorpioborn) [reply proto/sge/house/ticket.proto]: done

File: proto/sge/house/ticket.proto [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+message DepositTicketPayload {
+  // kyc_data contains the details of user kyc.
+  KycDataPayload kyc_data = 1 [ (gogoproto.nullable) = false ];
+}
+
+// WithdrawTicketPayload indicates data of bet withdrawal ticket.
```
[the remark this answers is not in the document]
Comment (scorpioborn) [reply proto/sge/house/ticket.proto]: done

File: x/house/client/cli/tx_deposit.go (hunk 1 above)
Comment (scorpioborn) [reply x/house/client/cli/tx_deposit.go:22]: done

File: x/house/client/cli/tx_withdraw.go (hunk 2 above)
Comment (scorpioborn) [reply x/house/client/cli/tx_withdraw.go:24]: done

File: x/house/types/ticket.go (hunk 3 above)
Comment (scorpioborn) [reply x/house/types/ticket.go]: done

**Review by 3eyed-raven — APPROVED** [review]

**Pull request merged**